### ISSUE TITLE
Stream Texture Import Bug from official examples

### ISSUE THREAD

**Issue by ManSoorSour** [KhronosGroup/UnityGLTF#462]
Im currently trying to test out the stream option but it seems like it doesn't import the textures correctly. If I run the official example scene "StreamTestScene", I get the following result:

![Test](https://user-images.githubusercontent.com/19176738/60528802-94454900-9cf5-11e9-9f98-6d9977e2a154.png)

The URI option works great, only the stream option gives me red/white question mark textures. The same bug appears for every model imported through the stream option.

**Comment by ManSoorSour** [comment]
GLB/GLTF-Binary-Models imported through the stream option works great. I actually don't get it, the only difference between the Stream & URI option is the ExternalDataLoader property from ImportOptions. But both looks fine for me... so why does the above bug only appear in stream options?

**Comment by Dryra** [comment]
Hello, 
I'm having the same issue, is there any news on this ?

**Comment by PWendtInneo** [comment]
Hi there, I just wanted to know, if this is still an issue. I had this Issue too, when it first came up, and I'm waiting for this to get solved before I'll update the plugin in my project.

**Comment by Stephengower** [comment]
To workaround this issue, in GLTFSceneImporter.cs, in the ConstructImageBuffer method, change the LoadStream call to use LoadStreamSync instead.

//  await _options.ExternalDataLoader.LoadStream(image.Uri);
 _options.ExternalDataLoader.LoadStreamSync(image.Uri);


For some reason, the async LoadStream method only works once, so only the first texture is loaded, and all others are empty.

**Comment by Dryra** [comment]
@Stephengower Thank you for the info! Saved me a lot of time! it works now as expected

**Comment by Stephengower** [comment]
Yeah, the current release is just flat out broken due to this, and it looks like it's been this way for a couple months at least.  Maybe all the devs took the summer off.

**Comment by Dryra** [comment]
I take the works as expected back, it's causing memory leaks on iOS.

**Comment by yaqinh** [comment]
[vr_table.zip](https://github.com/KhronosGroup/UnityGLTF/files/3632795/vr_table.zip)
Hello, I still see this issue. The texture is wrong for some files I have tried. (See the attachment GLTF file). @Stephengower 's method resolves this bug but is it the official way to do it? Hope the dev team could fix this ASAP. Thanks!

**Comment by Dryra** [comment]
@yaqinh No it is not, and it does not really fix the issue completely as it causes memory leaks on iOS. I guess we have to wait until the devs are back.

**Comment by yaqinh** [comment]
@Dryra , yeah, my platform is UWP and I have not found any issue yet. But, yeah memory leak on IOS is important

**Comment by fagerbur** [comment]
From what I can see, the issue stems from attempting to use File.OpenRead asynchronously when it wasn't created to natively work that way, at least initially. I don't have an iOS device to test this issue on so I can't reproduce the memory leak but by modifying the LoadFileStream method in FileLoader.cs to look like this below, the textures now load correctly.

```
private Task LoadFileStream(string rootPath, string fileToLoad)
{
	string pathToLoad = Path.Combine(rootPath, fileToLoad);
	if (!File.Exists(pathToLoad))
	{
		throw new FileNotFoundException("Buffer file not found", fileToLoad);
	}

	Task fileOpenTask = Task.Run(() => { LoadedStream = File.OpenRead(pathToLoad); });
	fileOpenTask.Wait();
	return Task.CompletedTask;
}
```

If someone can confirm whether this fixes the issue, I'll go ahead and make a pull request and see if i can get it merged in.

**Comment by fagerbur** [comment]
@AdamMitchell-ms Are you ok with the above implementation to resolve this issue? This also would resolve issue #484

**Comment by fagerbur** [comment]
this has been resolved by @mhochk in #532 and can be closed
